exec/java-exec/src/main/java/org/apache/drill/exec/rpc/user/UserSession.java

```diff
@@ -278,7 +277,7 @@ public boolean setSessionOption(String name, String value) {
    */
   public String registerTemporaryTable(AbstractSchema schema, String tableName) throws IOException {
       addTemporaryLocation((WorkspaceSchemaFactory.WorkspaceSchema) schema);
-      String temporaryTableName = Paths.get(sessionId, UUID.randomUUID().toString()).toString();
+      String temporaryTableName = new Path(sessionId, UUID.randomUUID().toString()).toUri().getPath();
```

Comment:
In this case we use path  just to concatenate two paths: session id + temp table generated name (to which refer as temp table full name). org.apache.hadoop.fs.Path uses common path delimiter which it can resolve for any system. The above just returns xxx/xxx without file system specifics.